src/physical/m_shuffle_lower.cpp

```diff
@@ -86,13 +86,18 @@ bool ShuffleLower::execute(Block *block) {
 				range_=buffer_->compare_start_end(ranges_2_);
 			if(range_==-1) continue;
 			if(shuffle_ser_obj_->exchange_id_!=10)
-				pcbuffer_->put(buffer_, range_);
+//				pcbuffer_->put(buffer_, range_);
+			/*
+			 * if put fail, then put it into a vector
+			 * then we can put all the vectors.
+			 *  */
+			while(!pcbuffer_->put(buffer_, range_));
```

Comment:
this is dangerous, because in particularly, this line can cause blocking, but it is almost impossible to happen, but this is a bug [notation--dangerous].